### PR TITLE
Update allowed-values docs and explain default value

### LINKED ISSUE #248: Clarify Default Value for Enumerations of Allowed Values Constraints for Metaschema Syntax
# User Story:

As a developer of current or future Metaschema-based tools, I would like to better understand the default behavior of `allowed-value` enumeration in `constraint`s, specifically when `allow-other` is not explicitly defined.

## Goals:

Clarify documentation for OSCAL developers to properly interpret the Metaschema definitions and/or implement Metaschema-aware tools for processing the primary source definitions of OSCAL.

## Dependencies:

N/A

## Acceptance Criteria

- [x] Correct the syntax and example for the "Enumerated Values" portion in the [Metaschema syntax change](https://pages.nist.gov/metaschema/specification/syntax/#enumerated-values).
- [x] Update the "Enumerated Values" portion implied default copy [in this section of the syntax page](https://pages.nist.gov/metaschema/specification/syntax/#enumerated-values).
- [x] All [website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes in this issue have been updated. Changes to the website can be made in the docs/content directory of your branch.
- [x] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [x] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### PR DESCRIPTION
**Pull request by aj-stein-nist**
# Committer Notes

Resolves #248.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- ~Have you written new tests for your core changes, as applicable?~
- ~Have you included examples of how to use your new feature(s)?~
- ~Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.~

### CODE REVIEW

**Comment by aj-stein-nist** [comment]
@david-waltermire-nist, per discussion today, this one's ready to go!

File: website/content/specification/syntax/_index.md [hunk 1]
```diff
@@ -646,23 +646,20 @@ Additionally, line feed (LF) characters must be escaped as "\\n" when appearing
 
 Additionally, flags may be constrained to a set of known values listed in advance.
 
-This restriction can be either strict (values must be in the list for document validity) or loose (i.e. for documentation only, no effect in schemas).
+This restriction can be either strict (values must be in the list for document validity with `allow-other="no"` attribute for an `allowed-values` element) or loose (i.e. for documentation only, no effect in schemas, with `allow-other="yes"`). If a `constraint` has an `allowed-values` element and the `allow-other` attribute is not explicitly defined, the default is `allow-other="no"`, validation is strict, and the only valid values are those in the list.
```
Comment (david-waltermire-nist) [inline website/content/specification/syntax/_index.md:649]:
```suggestion
This restriction can be either:

1. strict (values must be in the list for document validity with `allow-other="no"` attribute for an `allowed-values` element) or
2. loose (i.e. for documentation only, no effect in schemas, with `allow-other="yes"`).

If an `allowed-values` constraint does not have the `allow-other` attribute defined, the default is `allow-other="no"`, resulting in strict validation where the only valid values are those in the list.
```

Comment (david-waltermire-nist) [reply website/content/specification/syntax/_index.md:649]: Trying to make this easier for reference by breaking it up. Thoughts?

**Review by david-waltermire-nist — CHANGES_REQUESTED** [review]
Looks good! Thanks!

File: website/content/specification/syntax/_index.md (hunk 1 above)
Comment (aj-stein-nist) [reply website/content/specification/syntax/_index.md:649]: I don't like it, I love it!

**Review by david-waltermire-nist — APPROVED** [review]

**Pull request merged**